src/lib.rs

```diff
@@ -30,7 +30,13 @@ pub struct TimeoutConnector<T> {
     write_timeout: Option<Duration>,
 }
 
-impl<T: Connect> TimeoutConnector<T> {
+impl<T> TimeoutConnector<T>
+where
+    T: Service<Uri> + Send,
+    T::Response: AsyncRead + AsyncWrite + Send + Unpin,
+    T::Future: Send + 'static,
+    T::Error: Into<BoxError>,
+{
```

Comment:
The `Connect` trait still depends on `http1`/`http2` features so I had to expand the trait bounds manually.